test/libyul/SyntaxTest.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "libyul/AsmAnalysis.h"
+#include "libyul/AsmAnalysisInfo.h"
+#include <libyul/AsmParser.h>
+
+#include "libyul/backends/wasm/WasmDialect.h"
+#include <libyul/backends/evm/EVMDialect.h>
+#include <libyul/backends/wasm/WasmDialect.h>
+
+#include "liblangutil/EVMVersion.h"
+#include <liblangutil/Exceptions.h>
+
+#include <test/libyul/SyntaxTest.h>
+
+using namespace std;
+using namespace solidity;
+using namespace solidity::util;
+using namespace solidity::langutil;
+using namespace solidity::yul::test;
+
+namespace
+{
+std::map<string const, yul::Dialect const& (*)(langutil::EVMVersion)> validDialects = {
```

Comment:
Can you use std::function? Also maybe this should rather be a static function of SyntaxTest, if that is possible.

Reply:
I don't see advantage in using std::function here and only disadvantages: More compilation time, more complexity.
I suggest we leave this like this for now and clean it up with #8244 (unifying the settings from yuloptimizertest and syntaxtest)